towerlib/towerlib.py

```diff
@@ -1997,24 +2034,707 @@ def schedules(self):
+                    project_ids.append(project.id)
+        return project_ids
+
+    def get_production_branch_project_id(self, scm_url):
+        """Get the id of the ansible tower project, which has the branch production as scm branch.
+
+        Args:
+            scm_url: the scm url of the production branch project.
+
+        Returns:
+            The id of the production branch project.
+
+        """
+        project_id = None
+        projects = self.get_all_projects()
+        for project in projects:
+            if project.scm_type == 'git':
+                if project.scm_url == scm_url and project.scm_branch == 'production':
+                    project_id = project.id
+                    break
+        return project_id
+
+    def get_playbooks_by_branch_name(self, scm_url, scm_branch):
+        """Get all the playbooks of the project with the given scm_url and branch name.
+
+        Args:
+            scm_url: the given scm url.
+            scm_branch: the given scm branch
+
+        Returns:
+            list: a list with the names of all the playbooks for the matching project.
+
+        """
+        playbooks = None
+        projects = self.get_all_projects()
+        for project in projects:
+            if project.scm_url == scm_url and project.scm_branch == scm_branch:
+                playbooks = project.playbooks
+                break
+        return playbooks
+
+    def get_production_branch_playbooks(self, scm_url):
+        """Get all the playbooks of the project, which has the branch production as scm branch.
+
+        Args:
+            scm_url: given scm url
+
+        Returns:
+            list: a list with the names of all the playbooks of production branch project.
+
+        """
+        projects = self.get_all_projects()
+        for project in projects:
+            if project.scm_url == scm_url and project.scm_branch == 'production':
+                playbooks = project.playbooks
+                return playbooks
+
+    def get_ansible_facts_by_host_id(self, host_id):
+        """Get the ansible_facts of the given host.
+
+        Args:
+            host_id: id of the host for which the method will return the ansible_facts.
+
+        Returns:
+            dict: ansible_facts as dictionary.
+
+        """
+        host = self.get_host_by_id(host_id)
+        return host.ansible_facts()
+
+    def get_all_inventories(self):
+        """
+        Get all the inventories of the ansible tower.
+
+        Returns:
+            list: list of inventories as EntityManager object.
+        """
+        inventories = [item for item in self.inventories]
+        return inventories
+
+    def get_all_hosts(self):
+        """Get all the hosts of the given ansible tower.
+
+        Returns:
+            list: list of all the hosts as EntityManager object.
+
+        """
+        hosts = [item for item in self.hosts]
+        return hosts
+
+    def get_all_credentials(self):
```

Comment:
This is not needed. Client side usage list(tower.credentials) achieves the same.